ui-src/lib/views/RecordingsView.js

```diff
@@ -0,0 +1,230 @@
+ * representation of recordings.
+ *
+ * @todo This should be decoupled!
+ *
+ * @type {Array} Array of column names
+ */
+const _columnOrder = [
+  'start',
+  'end',
+  'resolution',
+  'frameRate',
+  'size',
+  'rate',
+];
+
+/**
+ * Labels for columns.
+ */
+const _columnLabels = {
+  start: 'Start',
+  end: 'End',
+  resolution: 'Resolution',
+  frameRate: 'FPS',
+  size: 'Storage',
+  rate: 'BitRate',
+};
+
+/**
+ * Class to encapsulate a view of a list of recordings from a single camera.
+ */
+export default class RecordingsView {
+  /**
+   * Construct display from camera data and use supplied formatter.
+   *
+   * @param  {Camera} camera camera object (immutable)
+   * @param  {RecordingFormatter} recordingFormatter Desired formatter
+   * @param  {Boolean} trimmed True if the display should include trimmed ranges
+   * @param  {jQuery} parent Parent to which new DOM is attached, or null
+   */
+  constructor(camera, recordingFormatter, trimmed = false, parent = null) {
+    this._cameraName = camera.shortName;
+    this._cameraRange = camera.range90k;
+    this._formatter = recordingFormatter;
+    this._element = $(`tab-${camera.uuid}`); // Might not be there initially
+    if (this._element.length == 0) {
+      this._element = this._createElement(camera.uuid, camera.shortName);
+    }
+    this._trimmed = trimmed;
+    this._recordings = null;
+    this._clickHandler = null;
+    if (parent) {
+      parent.append(this._element);
+    }
+    this._timeoutId = null;
+  }
+
+  /**
+   * Create DOM for the recording.
+   *
+   * @param {String} id DOM id for the main element
+   * @param {String} cameraName Name of the corresponding camera
+   * @return {jQuery} Partial DOM as jQuery object
+   */
+  _createElement(id, cameraName) {
+    const tab = $('<tbody>').attr('id', id);
+    tab.append(
+      $('<tr class="name">').append($('<th colspan=6/>').text(cameraName)),
+      $('<tr class="hdr">').append(
+      $(
+        _columnOrder
+          .map((name) => '<th>' + _columnLabels[name] + '</th>')
+          .join('')
+      )),
+      $('</tr>'),
+      $('<tr class="loading"><td colspan=6>loading...</td></tr>').hide()
+    );
+    return tab;
+  }
+
+  /**
+   * Update display for new recording values.
+   *
+   * Each existing row is reformatted.
+   *
+   * @param  {Array} newRecordings
+   * @param  {Boolean} trimmed  True if timestamps should be trimmed
+   */
+  _updateRecordings() {
+    const trimRange = this._trimmed ? this._cameraRange : null;
+    const recordings = this._recordings;
+    this._element.children('tr.r').each((rowIndex, row) => {
+      const values = this._formatter.format(recordings[rowIndex], trimRange);
+      $(row)
+        .children('td')
+        .each((index, element) => $(element).text(values[_columnOrder[index]]));
+    });
+  }
+
+  /**
+   * get whether time ranges in the recording list are being trimmed.
+   *
+   * @return {Boolean} [description]
```

Comment:
[description]?

Reply:
This, and several other instances fixed.

Reply:
fwiw, I just looked at [the style guide](https://google.github.io/styleguide/jsguide.html#jsdoc-method-and-function-comments), and it says you can omit stuff that's totally obvious. (It explicitly says that the type `{Boolean}` is not obvious, but you could just do `@return {Boolean}` if you prefer. I personally hate having to write essentially the same thing in N places, or even having to read it...

Reply:
OK, fixed in many places.